EqSat/src/UnionFind.cpp

```diff
@@ -24,12 +26,40 @@ Id UnionFind::find(Id id) const
     return id;
 }
 
+Id UnionFind::find(Id id)
+{
+    LUAU_ASSERT(size_t(id) < parents.size());
+
+    Id set = const_cast<const UnionFind*>(this)->find(id);
```

Comment:
One last thing: can you add a private `Id canonicalize(Id id) const` which takes the implementation of `Id find(Id id) const` and have `find` just forward to `canonicalize`? Then you don't need this `const_cast`. :)